checker/src/main/java/org/checkerframework/checker/resourceleak/ResourceLeakVisitor.java

```diff
@@ -476,4 +477,60 @@ private void checkOwningField(Element field) {
   protected boolean isWpiEnabledForRLC() {
     return enableWpiForRlc;
   }
+
+  /**
+   * Determine if the given expression <code>e</code> refers to <code>this.field</code>.
+   *
+   * @param e the expression
+   * @param field the field
+   * @return true if <code>e</code> refers to <code>this.field</code>
+   */
+  private boolean expressionEqualsField(String e, Element field) {
+    // TODO: this is very wrong
+    return e.contains(field.getSimpleName().toString());
```

Comment:
Big oof. Thanks for catching that the prior code was doing this - not our finest moment.

The correct thing to do here is probably to convert `e` into a Java expression using one of the methods in `StringToJavaExpression`, and then make sure that the result represents the same field as `field` somehow.

Leaving this as a TODO is probably okay, but we should probably also open an issue. I'll try to do that (and scare @msridhar) as soon as I can.

Reply:
https://github.com/typetools/checker-framework/issues/6276